### PR TITLE
feat: support OAuth2 OIDC login via okta

### PR DESCRIPTION
**Pull request by n1ru4l**
This allows setting up login via Okta for self-hosted apps.

~~In addition to that this is a prerequisite for applying to the Okta Integration Network - which allows us to support multi-tenancy on the Hive cloud version. (See https://github.com/kamilkisiela/graphql-hive/issues/468#issuecomment-1283555362)~~ **EDIT** Organization level OIDC integration is added here: https://github.com/kamilkisiela/graphql-hive/pull/524

Configuration Documentation for the Okta App

<details>
  <summary>Create App</summary>

  image

</details>

<details>
  <summary>LogIn via Okta App</summary>

  image

</details>

<details>
  <summary>Okta API Scopes</summary>

  image

</details>

### CODE REVIEW

**Comment by n1ru4l** [comment]
Docs for this has been added in https://github.com/kamilkisiela/graphql-hive/pull/511

File: packages/web/docs/pages/self-hosting/oidc-login.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+For the `Sign-out redirect URIs` option full in the web app base url followed by `/auth/logout` (e.g. `https://app.graphql-hive.com/logout`).
+
+For the `Trusted Origins` option full in the web app base url (e.g. `https://app.graphql-hive.com`).
+
+![New web app integration](../../public/docs/pages/self-hosting/oidc-login/okta-new-web-app-integration.png)
+
+Afterwards, open the `General` tab of the created app and adjust the `General settings` as follows:
+
+In the `LOGIN` section, set the `Login initiated by` field to `Euther Okta or App`.
```
Comment (dotansimha) [inline packages/web/docs/pages/self-hosting/oidc-login.mdx:57]: typo: `Euther`

File: deployment/index.ts [hunk 2]
```diff
@@ -188,6 +188,12 @@ const googleConfig = {
   clientSecret: oauthConfig.requireSecret('googleSecret'),
 };
 
+const oktaConfig = {
+  clientId: oauthConfig.requireSecret('oktaClient'),
```
Comment (dotansimha) [inline deployment/index.ts:192]: if Otka login is not relevant for the SaaS service, I guess this one should be just `getSecret` and not `requireSecret`?

File: deployment/services/app.ts [hunk 3]
```diff
@@ -46,6 +47,11 @@ export function deployApp({
     clientId: pulumi.Output<string>;
     clientSecret: pulumi.Output<string>;
   };
+  oktaConfig: {
```
Comment (dotansimha) [inline deployment/services/app.ts:50]: this one can be optional?

File: deployment/index.ts (hunk 2 above)
Comment (n1ru4l) [reply deployment/index.ts:192]: I kind of added it here so we still have a way of testing it in prod 🤔 But you are right, we can make it `getSecret` - @kamilkisiela what do you think?

File: packages/web/docs/pages/self-hosting/oidc-login.mdx (hunk 1 above)
Comment (n1ru4l) [reply packages/web/docs/pages/self-hosting/oidc-login.mdx]: https://github.com/kamilkisiela/graphql-hive/pull/510/commits/070e4128bf876855aae92b7c8b95e273d4d4f6e7

**Comment by n1ru4l** [comment]
will be implemented as part of https://github.com/kamilkisiela/graphql-hive/pull/524

**Pull request closed without merging**